oncrpc4j-core/src/main/java/org/dcache/xdr/RpcCall.java

```diff
@@ -341,16 +352,18 @@ public void call(int procedure, XdrAble args, CompletionHandler<RpcReply, XdrTra
      * @param procedure The number of the procedure.
      * @param args The argument of the procedure.
      * @param type The expected type of the reply
+     * @param timeoutValue timeout value. 0 means no timeout
+     * @param timeoutUnits units for timeout value
      * @return A Future representing the result of the operation.
      * @throws OncRpcException
      * @throws IOException
      * @since 2.4.0
      */
-    public <T extends XdrAble> Future<T> call(int procedure, XdrAble args, final Class<T> type)
-            throws OncRpcException, IOException {
+    public <T extends XdrAble> Future<T> call(int procedure, XdrAble args, final Class<T> type, long timeoutValue, TimeUnit timeoutUnits)
```

Comment:
does this method really makes sense? The Future already has a get with timeout. I think

call(int procedure, XdrAble args, CompletionHandler<RpcReply, XdrTransport> callback, long timeoutValue, TimeUnit timeoutUnits)

is sufficient. Especially it CompletionHandler#fail will be called with TimeoutException.

Do you have some example in mind?